src/subdomains/supporting/payin/strategies/send/impl/solana-token.strategy.ts

```diff
@@ -35,9 +35,14 @@ export class SolanaTokenStrategy extends SolanaStrategy {
   }
 
   protected async prepareSend(payIn: CryptoInput, nativeFee: number): Promise<void> {
-    const prepareTxId = await this.topUpCoin(payIn, nativeFee);
+    const createTokenFee = await this.payInSolanaService.getCreateTokenAccountFee(
+      payIn.destinationAddress.address,
+      payIn.asset,
+    );
+
+    const feeAmount = nativeFee + createTokenFee;
```

Comment:
I think this calculation should be a level above so the total amount get's stored in the pay-in.